src/trans.rs

```diff
@@ -331,41 +353,165 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                 self.fun_types.insert(self.sig.clone(), ty);
             }
 
+            // Create the function prologue
+            // TODO: the epilogue and prologue are not always necessary
+            let read_sp = BinaryenLoad(self.module, 4, 0, 0, 0, BinaryenInt32(),
+                                       BinaryenConst(self.module, BinaryenLiteralInt32(0)));
```

Comment:
yes, I already started doing this a bit (surely not enough) for the code that now compiles, and this was also why I was asking on IRC about run-passes so we can test the interpreted output as well.